### PR TITLE
add dynamic grant section with local tests

### PR DESCRIPTION
**Pull request by yk12OT**

### CODE REVIEW

**Comment by yk12OT** [comment]
### Local tests
```
$ terraform plan
var.serviceregistry_api
  Service registry API endpoint to test against

  Enter a value: 
```

With Production `serviceregistry API endpoint` the following results are produced:
```
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

module.s3_test_acl_grants.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_bucket_with_team.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_bucket_with_team.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_acl_grants.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_bucket_with_app.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_bucket_encryption.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_bucket_with_app.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_bucket_encryption.module.team_org.data.http.team_org: Refreshing state...

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.s3_test_acl_grants.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-acl-grants"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket, please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + grant {
          + permissions = [
              + "READ",
            ]
          + type        = "Group"
          + uri         = "http://acs.amazonaws.com/groups/global/AllUsers"
        }
      + grant {
          + permissions = [
              + "READ_ACP",
              + "WRITE",
            ]
          + type        = "Group"
          + uri         = "http://acs.amazonaws.com/groups/s3/LogDelivery"
        }
      + grant {
          + id          = "ae91c3bb39136d843c2dfe68f2ef07d6413b83ce9c00782fccf5c866e1ca5f9f"
          + permissions = [
              + "FULL_CONTROL",
            ]
          + type        = "CanonicalUser"
        }

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_encryption.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-encryption"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket, please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + server_side_encryption_configuration {
          + rule {
              + apply_server_side_encryption_by_default {
                  + sse_algorithm = "AES256"
                }
            }
        }

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_with_app.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-with-app-tag"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "app"       = "echo"
          + "concern"   = "This is a test bucket, please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_with_team.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-with-team-tag"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket, please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

Plan: 4 to add, 0 to change, 0 to destroy.

-------------------------------------------------------------------------
```

**Comment by yk12OT** [comment]
### Local tests
```
$ terraform plan
var.serviceregistry_api
  Service registry API endpoint to test against

  Enter a value: 
```

With Production `serviceregistry API endpoint` the following results are produced:
```
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

module.s3_test_bucket_with_app.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_bucket_with_app.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_bucket_different_region.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_acl_grants.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_bucket_different_region.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_acl_grants.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_bucket_with_team.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_bucket_encryption.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_bucket_encryption.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_bucket_with_team.module.ownership.data.http.service_owner: Refreshing state...

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.s3_test_acl_grants.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-acl-grants"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket, please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + grant {
          + permissions = [
              + "READ",
            ]
          + type        = "Group"
          + uri         = "http://acs.amazonaws.com/groups/global/AllUsers"
        }
      + grant {
          + permissions = [
              + "READ_ACP",
              + "WRITE",
            ]
          + type        = "Group"
          + uri         = "http://acs.amazonaws.com/groups/s3/LogDelivery"
        }
      + grant {
          + id          = "some_aws_canonic_user_id"
          + permissions = [
              + "FULL_CONTROL",
            ]
          + type        = "CanonicalUser"
        }

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_different_region.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-different-region"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-west-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket, please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_encryption.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-encryption"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-west-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket, please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + server_side_encryption_configuration {
          + rule {
              + apply_server_side_encryption_by_default {
                  + sse_algorithm = "AES256"
                }
            }
        }

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_with_app.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-with-app-tag"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "app"       = "echo"
          + "concern"   = "This is a test bucket, please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_with_team.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-with-team-tag"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket, please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

Plan: 5 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------
```

File: tests/test_s3_with_policy.tf [hunk 1]
```diff
@@ -0,0 +1,47 @@
+module "s3_test_with_policy" {
+  source              = "../modules/s3"         # local testing by a developer
+  serviceregistry_api = var.serviceregistry_api # enter a service registry URI
+
+  bucket = "od-test-with-policy"
+  region = var.region
+
+  policy = replace(local.s3_public_read, "BUCKET", "od-test-with-policy")
```
Comment (leochu) [inline tests/test_s3_with_policy.tf]: The policy document can be externalized as a template file.  
https://www.terraform.io/docs/providers/template/d/file.html

File: tests/variables.tf [hunk 2]
```diff
@@ -0,0 +1,84 @@
+variable "serviceregistry_api" {
+  type        = string
+  description = "Service registry API endpoint to test against"
+
+  default = "https://serviceregistry-production-http.managed.services.opendoor.com"
```
Comment (leochu) [inline tests/variables.tf]: Remove this before publishing this repo?

File: tests/test_s3_with_policy.tf (hunk 1 above)
Comment (leochu) [reply tests/test_s3_with_policy.tf]: Not a blocker

Comment (yk12OT) [reply tests/test_s3_with_policy.tf]: DONE, thanks for the suggestion! It also helped to fix other problems like this.

File: tests/variables.tf (hunk 2 above)
Comment (yk12OT) [reply tests/variables.tf]: done

**Comment by yk12OT** [comment]
### Local tests
```
$ terraform plan
var.serviceregistry_api
  Service registry API endpoint to test against

  Enter a value: 
```

With Production `serviceregistry API endpoint` the following results are produced:
```
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

module.s3_test_bucket_cores.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_versioning.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_bucket_encryption.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_lifecycle.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_with_policy.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_lifecycle.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_with_policy.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_acl_grants.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_bucket_different_region.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_bucket_encryption.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_bucket_cores.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_acl_grants.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_bucket_with_app.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_versioning.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_bucket_with_team.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_bucket_different_region.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_logging.module.team_org.data.http.team_org: Refreshing state...
module.s3_test_bucket_with_app.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_logging.module.ownership.data.http.service_owner: Refreshing state...
module.s3_test_bucket_with_team.module.team_org.data.http.team_org: Refreshing state...

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.s3_test_acl_grants.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-acl-grants"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket: please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + grant {
          + permissions = [
              + "READ",
            ]
          + type        = "Group"
          + uri         = "http://acs.amazonaws.com/groups/global/AllUsers"
        }
      + grant {
          + permissions = [
              + "READ_ACP",
              + "WRITE",
            ]
          + type        = "Group"
          + uri         = "http://acs.amazonaws.com/groups/s3/LogDelivery"
        }

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_cores.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-cors-rules"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket: please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + cors_rule {
          + allowed_headers = [
              + "*",
            ]
          + allowed_methods = [
              + "GET",
            ]
          + allowed_origins = [
              + "*",
            ]
          + expose_headers  = []
          + max_age_seconds = 3000
        }

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_different_region.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-different-region"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-west-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket: please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_encryption.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-encryption"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket: please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + server_side_encryption_configuration {
          + rule {
              + apply_server_side_encryption_by_default {
                  + sse_algorithm = "AES256"
                }
            }
        }

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_with_app.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-with-app-tag"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "app"       = "echo"
          + "concern"   = "This is a test bucket: please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_bucket_with_team.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-with-team-tag"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket: please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_lifecycle.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-lifecycle"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket: please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + lifecycle_rule {
          + abort_incomplete_multipart_upload_days = 0
          + enabled                                = true
          + id                                     = "expire-and-delete-after-30-days"

          + expiration {
              + days                         = 30
              + expired_object_delete_marker = false
            }

          + noncurrent_version_expiration {
              + days = 1
            }
        }
      + lifecycle_rule {
          + abort_incomplete_multipart_upload_days = 0
          + enabled                                = true
          + id                                     = "S3 transition to INTELLIGENT_TIERING"

          + noncurrent_version_transition {
              + days          = 3
              + storage_class = "INTELLIGENT_TIERING"
            }

          + transition {
              + days          = 3
              + storage_class = "INTELLIGENT_TIERING"
            }
        }

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_logging.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-logging"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket: please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + grant {
          + permissions = [
              + "READ_ACP",
              + "WRITE",
            ]
          + type        = "Group"
          + uri         = "http://acs.amazonaws.com/groups/s3/LogDelivery"
        }

      + logging {
          + target_bucket = "od-test-logging"
          + target_prefix = "logs/"
        }

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

  # module.s3_test_versioning.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-versioning"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket: please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + versioning {
          + enabled    = true
          + mfa_delete = false
        }
    }

  # module.s3_test_with_policy.aws_s3_bucket.s3_bucket will be created
  + resource "aws_s3_bucket" "s3_bucket" {
      + acceleration_status         = (known after apply)
      + acl                         = "private"
      + arn                         = (known after apply)
      + bucket                      = "od-test-with-policy"
      + bucket_domain_name          = (known after apply)
      + bucket_regional_domain_name = (known after apply)
      + force_destroy               = false
      + hosted_zone_id              = (known after apply)
      + id                          = (known after apply)
      + policy                      = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "s3:GetObject"
                      + Effect    = "Allow"
                      + Principal = "*"
                      + Resource  = "arn:aws:s3:::od-test-with-policy/*"
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + region                      = "us-east-2"
      + request_payer               = (known after apply)
      + tags                        = {
          + "Terraform" = "github.com/opendoor-labs/od-terraform/tests"
          + "concern"   = "This is a test bucket: please ignore"
          + "env"       = "staging"
          + "org"       = "infra"
          + "team"      = "cloud-infra"
        }
      + website_domain              = (known after apply)
      + website_endpoint            = (known after apply)

      + versioning {
          + enabled    = (known after apply)
          + mfa_delete = (known after apply)
        }
    }

Plan: 10 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------
```

**Pull request merged**